src/frontend/components/Button/index.tsx

```diff
@@ -0,0 +1,110 @@
+import React from 'react';
+import { Box, Button as GrommetButton, Text } from 'grommet';
+import { normalizeColor } from 'grommet/utils';
+
+import { SvgProps } from 'components/SVGIcons';
+import { theme } from 'utils/theme/theme';
+import styled from 'styled-components';
+
+const StyledGrommetButton = styled(GrommetButton)`
+  height: 100%;
+  padding: 0;
+`;
+
+const StyledBox = styled(Box)`
+  aspect-ratio: 1;
+  margin: 0 8px;
+  min-height: 100%;
+  position: relative;
+`;
+
+const StyledText = styled(Text)`
+  white-space: nowrap;
+`;
+
+interface ButtonProps {
+  badge?: JSX.Element;
+  disabled?: boolean;
+  Icon?: React.FC<SvgProps>;
+  label?: string;
+  onClick?: () => void;
+  reversed?: boolean;
+  title?: string;
+}
+
+export const Button = ({
+  badge,
+  disabled,
+  Icon,
+  label,
+  onClick,
+  reversed,
+  title,
+}: ButtonProps) => {
+  return (
+    <StyledGrommetButton
+      a11yTitle={title}
+      disabled={disabled}
+      onClick={onClick}
+      plain
+      title={title}
+    >
+      {({ hover }) => {
+        let tintColor;
+        if (disabled) {
+          tintColor = normalizeColor('blue-off', theme);
+        } else if (reversed) {
+          tintColor = normalizeColor(
+            hover ? 'blue-active' : 'blue-focus',
+            theme,
+          );
+        } else {
+          tintColor = normalizeColor(
+            hover ? 'blue-focus' : 'blue-active',
+            theme,
+          );
+        }
+
+        let iconColor;
+        let containerStyle;
+
+        if (
+          (!!reversed && (!hover || disabled)) ||
+          (!reversed && hover && !disabled)
+        ) {
+          iconColor = normalizeColor('white', theme);
+          containerStyle = {
+            background: tintColor,
+            borderRadius: '6px',
+            marginBottom: '6px',
+          };
+        } else {
+          iconColor = tintColor;
+          containerStyle = {
+            background: 'none',
+            marginBottom: '6px',
+          };
+        }
+
+        return (
+          <Box align={'center'} height={'100%'} width={'100%'}>
+            <StyledBox flex={{ grow: 1 }}>
+              {Icon && (
+                <Icon containerStyle={containerStyle} iconColor={iconColor} />
+              )}
+              {badge}
+            </StyledBox>
+            {label && (
+              <StyledText
+                size="12px"
+                color={normalizeColor('blue-active', theme)}
+              >
+                {label}
+              </StyledText>
+            )}
+          </Box>
+        );
+      }}
```

Comment:
If you are writing js logic with variables and return statement within JSX, that means its time to move this logic into a new component.

Reply:
this is done in next PR since I need to use the layout without the button component, is it ok for you ?

Reply:
ok :+1: